### PR TITLE
Implement csvTransactions

### PR DESCRIPTION
**Pull request by samerbuna**

### CODE REVIEW

File: src/graphql/types/object/business-account.ts [hunk 1]
```diff
@@ -37,10 +40,25 @@ const BusinessAccount = new GT.Object({
...
-        walletIds: {
-          type: GT.NonNullList(GT.ID),
+        walletNames: {
+          type: GT.NonNullList(WalletName),
         },
       },
+      resolve: async (source, args) => {
+        const walletIds: WalletId[] = []
+
+        for (const id of source.walletIds) {
```
Comment (bodymindarts) [inline src/graphql/types/object/business-account.ts:50]: This should be implemented as a use case method under `app` _not_ directly in the resolver!

File: src/graphql/types/object/consumer-account.ts [hunk 2]
```diff
@@ -43,10 +45,25 @@ const ConsumerAccount = new GT.Object({
...
-        walletIds: {
-          type: GT.NonNullList(GT.ID),
+        walletNames: {
+          type: GT.NonNullList(WalletName),
         },
       },
+      resolve: async (source, args) => {
+        const walletIds: WalletId[] = []
+
+        for (const id of source.walletIds) {
```
Comment (bodymindarts) [inline src/graphql/types/object/consumer-account.ts:55]: This looks like duplication - should be calling a use-case method not live in the resolver.

File: src/core/wallet-factory.ts [hunk 3]
```diff
@@ -43,3 +45,11 @@ export const getWalletFromRole = async ({ logger, role }) => {
   const user = await User.findOne({ role })
   return WalletFactory({ user, logger })
 }
+
+export const getStringCsvForWallets = async (walletIds: WalletId[]) => {
```
Comment (bodymindarts) [inline src/core/wallet-factory.ts:49]: Why is new code being added under `core`? This is fragmenting the code base. The code you have here + under the resolver should be unified under an `app/accounts` use case.

**Review by bodymindarts — CHANGES_REQUESTED** [review]

File: src/app/accounts/index.ts [hunk 4]
```diff
@@ -24,3 +25,26 @@ export const getBusinessMapMarkers = async () => {
...
+  const walletIds: WalletId[] = []
+
+  for (const walletName of walletNames) {
+    const wallet = await wallets.findByWalletName(walletName)
+    if (wallet instanceof Error) {
+      return wallet
+    }
+
+    if (!account.walletIds.includes(wallet.id)) {
+      return new RepositoryError()
```
Comment (dolcalmi) [inline src/app/accounts/index.ts:44]: Repository error is an error that only should be used at repository level. Please use a custom error, AuthorizationError, ValidationError

**Review by dolcalmi — CHANGES_REQUESTED** [review]
I did the review assuming that this is not a clean up

File: src/app/accounts/index.ts (hunk 4 above)
Comment (samerbuna) [reply src/app/accounts/index.ts]: Done

File: src/graphql/types/object/business-account.ts (hunk 1 above)
Comment (samerbuna) [reply src/graphql/types/object/business-account.ts]: Done

File: src/graphql/types/object/consumer-account.ts (hunk 2 above)
Comment (samerbuna) [reply src/graphql/types/object/consumer-account.ts]: Done

File: src/core/wallet-factory.ts (hunk 3 above)
Comment (samerbuna) [reply src/core/wallet-factory.ts]: Done

**Review by dolcalmi — APPROVED** [review]

File: src/app/accounts/index.ts [hunk 5]
```diff
@@ -24,3 +25,26 @@ export const getBusinessMapMarkers = async () => {
   const accounts = AccountsRepository()
   return accounts.listBusinessesForMap()
 }
+
+export const toWalletIds = async (
```
Comment (bodymindarts) [inline src/app/accounts/index.ts:29]: Just a minor thing - I think we could find a better name here - perhaps `filteredWalletIds` or something like that.

**Review by bodymindarts — APPROVED** [review]
Will approve but have a small rename suggestion - can fix or merge as you want.

File: src/app/accounts/index.ts (hunk 5 above)
Comment (samerbuna) [reply src/app/accounts/index.ts:29]: Will include that in the new conversion

**Pull request merged**